packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
+    for (const [nestedStackLogicalId, resource] of parentNestedStackResources) {
+      if (!this.isCdkManagedNestedStack(resource)) {
+        continue;
+      }
+
+      const nestedStackResource: NestedStackResource = resource;
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResources, sdk,
+      );
+
+      const populatedNestedTemplates = this.populateParentTemplates(
+        generatedParentTemplate.Resources[nestedStackLogicalId],
+        (deployedParentTemplate.Resources ?? {})[nestedStackLogicalId],
+        nestedStackTemplates,
+      );
+
+      await this.replaceNestedStacksInParentTemplate(
+        rootStackArtifact,
+        populatedNestedTemplates.generatedNestedTemplate,
+        populatedNestedTemplates.deployedNestedTemplate,
+        nestedStackTemplates.nestedStackName,
+        sdk,
+      );
+    }
+  }
+
+  private async getNestedStackTemplates(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
+    parentStackName: string | undefined, listStackResources: ListStackResources | undefined, sdk: ISDK,
+  ): Promise<NestedStackTemplates> {
+    const nestedTemplatePath = path.join(rootStackArtifact.assembly.directory, nestedTemplateAssetPath);
+    // CFN generates the nested stack name in the form `ParentStackName-NestedStackLogicalID-SomeHashWeCan'tCompute,
+    // so we get the ARN and manually extract the name.
+    //const nestedStackArn = (await listStackResources?.listStackResources())?.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+    let nestedStackArn: string | undefined = undefined;
+
+    try {
+      const stackResources = await listStackResources?.listStackResources();
+
+      if (stackResources) {
+        nestedStackArn = stackResources.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+      }
+    } catch (e) {
+      if (e.message !== `Stack with id ${parentStackName} does not exist`) {
+        throw e;
+      }
+    }
+    const nestedStackName = nestedStackArn?.slice(nestedStackArn.indexOf('/') + 1, nestedStackArn.lastIndexOf('/'));
+
+    return {
+      generatedNestedTemplate: JSON.parse(fs.readFileSync(nestedTemplatePath, 'utf-8')),
+      deployedNestedTemplate: nestedStackName
+        ? await this.readCurrentStackTemplate(rootStackArtifact, nestedStackName, sdk) : { Resources: {} },
+      nestedStackName,
+    };
+  }
+
+  private isCdkManagedNestedStack(stackResource: any): stackResource is NestedStackResource {
+    return stackResource.Type === 'AWS::CloudFormation::Stack' && stackResource.Metadata && stackResource.Metadata['aws:asset:path'];
+  }
+
+  private populateParentTemplates(
```

Comment:
I have to say, I don't understand the purpose of this method at all 😛.

It takes a `generatedNestedTemplate` and `deployedNestedTemplate` as arguments, but _also_ a `nestedStackTemplates`, and `NestedStackTemplates` has properties called `generatedNestedTemplate` and `deployedNestedTemplate`!

This is hurting my poor little brain 🙂.

Reply:
This method replaces the nested stack object in `generatedNestedTemplate` and `deployedNestedTemplate` with the full nested templates that live in `nestedStackTemplates`.

It was supposed to be a helper method to shorten some variable names and prevent giant expressions like `generatedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate` by doing the actual replacement in the helper method. The idea was returning the updated templates would let us refer to that long expression above without having to spell it out by encoding it as a return value.